receiver/otlpreceiver/config.go

```diff
@@ -32,8 +32,8 @@ const (
 
 // Protocols is the configuration for the supported protocols.
 type Protocols struct {
-	GRPC *configgrpc.GRPCServerSettings `mapstructure:"grpc"`
-	HTTP *confighttp.HTTPServerSettings `mapstructure:"http"`
+	GRPC *configgrpc.GRPCServerSettings `mapstructure:"grpc,omitempty"`
+	HTTP *confighttp.HTTPServerSettings `mapstructure:"http,omitempty"`
```

Comment:
do we really need omitempty on pointers? Also why do we need omitempty in this case, and in general, is not documented.

Reply:
If `omitempty` isn't included, the marshalled config will include a `grpc` or `http` field even for nil pointers. It'll marshal as `grpc: {}` or `http: {}`, which is fine and valid, but unnecessarily clutters up the config. Don't see a point in including the field if it isn't being used and isn't set.

Reply:
let's not add more API surface if not needed. I would avoid having `omitempty ` until we are clear that we need it.

Reply:
Ping on this.

Reply:
Sorry. Must've forgotten to remove this.